### ISSUE TITLE
[Safari] added_udm session flag prevents it from working on repeated searches in same tab

### ISSUE THREAD

**Issue by alexkli** [ZenithO-o/Fix-Google-Web-Search#1]
The `added_udm` flag prevents the script from applying udm=14 on subsequent searches in the same tab **when entering a new search in the address bar** (ie. not in Google's search field).

I am not sure if this is intended or just happens in Safari. I am using the script in Safari with [Userscripts](https://apps.apple.com/us/app/userscripts/id1463298887) extension.

It works for me well if I remove the `added_udm` logic from the script.

**Comment by isaacl** [comment]
I can confirm this issue affects safari.   It also seems to affect chrome when using omnibox to search.

Maybe the script should both look at session store as well as ensure the utm param is already in the url?  idk

**Comment by alexkli** [comment]
I think it’s enough to check if it’s on the URL. It doesn’t look like Google remembers it, and even if, always setting it on the URL should do no harm(?).

**Comment by isaacl** [comment]
Always setting it isn't ideal though, because you get a redirect every search.

**Comment by isaacl** [comment]
Yeah I think best course is to set if it's there's no utm search param and drop the session storage approach (which I really liked, it just doesn't seem to work)

**Comment by ZenithO-o** [comment]
Hi Alexander, thanks for leaving an issue request. I see what both you and Issac are saying about subsequent searches using the address bar. Originally used session storage so that you could switch to All search and prevent redirection back to Web search, but it turns out not to be a clean solution. :(

I looked into how the `udm` parameter is used and it's fascinating... though really annoying to handle. Only *some* features use it (i.e. Images & Web), while others don't (All, Videos, News, Shopping, etc.). Not really in the business of reverse engineering Google's search any further, but a quick solution seems to be to add a dummy param like `udm=1` to those links.


**TL;DR:** New script handles searches from search bar + retains mode of search when using UI. 

Let me know what you think!

**Comment by alexkli** [comment]
Just tested updated script seems to work fine 👍 Thanks!
